posit_workbench_best_practice_with_r.md

```diff
@@ -0,0 +1,485 @@
+
+## Coding with R
+
+### Removing intermediate variables
+
+Suppose we have two large data frames, `a_data` and `b_data`, that we join together to create a third data frame, `combined_data` that we will use for the rest of our analysis. Our environment will now have 3 large data frames stored in memory, even though `a_data` and `b_data` may now be redundant. We can free up that occupied memory by deleting `a_data` and `b_data` from the environment using the `rm()` function:
+
+```r
+rm(a_data, b_data)
+```
+
+This allows R to use that memory to store other potentially large objects without running out of memory.
+
+R will automatically perform _garbage collection_ — the process of freeing-up unused system memory that's no longer used — where necessary. But we can also manually trigger garbage collection by running `gc()` in the R console at any time, which can be especially effective after large objects have just been deleted from the environment.
+
+### Overwriting existing objects
+
+It's quite common to modify existing objects on a step-by-step basis. One strategy for reducing the amount of variables you create in your environment is to modify and overwrite the existing object, by saving the result into the same object.
+
+Let's imagine we are working with the iris dataset:
+
+```r
+data("iris")
+head(iris)
+```
+
+```
+  Sepal.Length Sepal.Width Petal.Length Petal.Width Species
+1          5.1         3.5          1.4         0.2  setosa
+2          4.9         3.0          1.4         0.2  setosa
+3          4.7         3.2          1.3         0.2  setosa
+4          4.6         3.1          1.5         0.2  setosa
+5          5.0         3.6          1.4         0.2  setosa
+6          5.4         3.9          1.7         0.4  setosa
+```
+
+and we want to approximate the area of the sepal and petal by multiplying "Length" × "Width", and convert the Species values to upper-case.
+
+When working with standard R methods, we could modify the existing iris by adding and modifying columns.
+
+```r
+data("iris")
+iris$Sepal.Area <- iris$Sepal.Length * iris$Sepal.Width
+iris$Petal.Area <- iris$Petal.Length * iris$Petal.Width
+iris$Species <- toupper(iris$Species)
+head(iris)
+```
+
+```
+  Sepal.Length Sepal.Width Petal.Length Petal.Width Species Sepal.Area
+1          5.1         3.5          1.4         0.2  SETOSA      17.85
+2          4.9         3.0          1.4         0.2  SETOSA      14.70
+3          4.7         3.2          1.3         0.2  SETOSA      15.04
+4          4.6         3.1          1.5         0.2  SETOSA      14.26
+5          5.0         3.6          1.4         0.2  SETOSA      18.00
+6          5.4         3.9          1.7         0.4  SETOSA      21.06
+  Petal.Area
+1       0.28
+2       0.28
+3       0.26
+4       0.30
+5       0.28
+6       0.68
+```
```

Comment:
I'd argue we shouldn't promote the use of *base* R?

Reply:
Agreed.